types/vue.d.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ComponentOptions,
+  WatchOptions,
+  WatchHandler,
+  DirectiveOptions,
+  DirectiveFunction
+} from "./options.d";
+import { VNode, VNodeData, VNodeChildren } from "./vnode.d";
+import { PluginFunction, PluginObject } from "./plugin.d";
+
+export declare class Vue {
+
+  constructor(options?: ComponentOptions);
+
+  $data: Object;
+  readonly $el: HTMLElement;
+  readonly $options: ComponentOptions;
+  readonly $parent: Vue;
+  readonly $root: Vue;
+  readonly $children: Vue[];
+  readonly $refs: { [key: string]: Vue };
+  readonly $slots: { [key: string]: VNode[] };
+  readonly $isServer: boolean;
+
+  $mount(elementOrSelector?: Element | String, hydrating?: boolean): this;
+  $forceUpdate(): void;
+  $destroy(): void;
+  $set: typeof Vue.set;
+  $delete: typeof Vue.delete;
+  $watch(
+    expOrFn: string | Function,
+    callback: WatchHandler,
+    options?: WatchOptions
+  ): (() => void);
+  $on(event: string, callback: Function): this;
+  $once(event: string, callback: Function): this;
+  $off(event?: string, callback?: Function): this;
+  $emit(event: string, ...args: any[]): this;
+  $nextTick(callback?: (this: this) => void): void;
+  $createElement(
+    tag?: string | Vue,
+    data?: VNodeData,
+    children?: VNodeChildren,
+    namespace?: string
+  ): VNode;
+
+
+  static config: {
+    silent: boolean;
+    optionMergeStrategies: any;
+    devtools: boolean;
+    errorHandler(err: Error, vm: Vue): void;
+    keyCodes: { [key: string]: number };
+  }
+
+  static extend(options: ComponentOptions): Vue;
+  static nextTick(callback: () => void, context?: any[]): void;
+  static set<T>(object: Object, key: string, value: T): T;
```

Comment:
In 2.0, Vue.set can also set array items.
https://github.com/vuejs/vue/blob/next/src/core/observer/index.js#L190
So, we can define this method as...

```js
static set<T>(object: Object, key: string, value: T): T;
static set<T>(array: T[], key: string, value: T): T;
```